sqlglot/optimizer/scope.py

```diff
@@ -509,7 +509,9 @@ def traverse_scope(expression: exp.Expression) -> t.List[Scope]:
     Returns:
         list[Scope]: scope instances
     """
-    if not isinstance(expression, exp.Unionable):
+    if isinstance(expression, exp.Create):
+        return list(_traverse_scope(Scope(expression)))
+    elif not isinstance(expression, exp.Unionable):
```

Comment:
Maybe we should make this a bit more generic by creating a DDLExpression class or something (Toby mentioned something like this in another comment) that `Create` and `Insert` inherit from, so that we can deal with both of them at the same time.